### PR TITLE
feat(APP-3310): Upgrade Storybook to v8

### PR DESCRIPTION
**Pull request by rcaragon**
## Description

Upgrade Storybook to v8.

Task: [APP-3310](https://aragonassociation.atlassian.net/browse/APP-3310)

## Checklist:

-   [x] I have selected the correct base branch.
-   [x] I have performed a self-review of my own code.
-   [x] I have commented my code, particularly in hard-to-understand areas.
-   [x] I have made corresponding changes to the documentation.
-   [x] My changes generate no new warnings.
-   [x] Any dependent changes have been merged and published in downstream modules.
-   [x] I ran all tests with success and extended them if necessary.
-   [x] I have updated the `CHANGELOG.md` file in the root folder of the package after the [UPCOMING] title and before
        the latest version.
-   [ ] I have tested my code on the test network.

[APP-3310]: https://aragonassociation.atlassian.net/browse/APP-3310?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -83,18 +83,21 @@
     "@babel/preset-env": "^7.24.7",
     "@babel/preset-react": "^7.24.7",
     "@babel/preset-typescript": "^7.24.7",
+    "@chromatic-com/storybook": "^1",
```
Comment (cgero-eth) [inline package.json:86]: I would remove the introduction of the chromatic tool for another ticket and discuss it first internally

Comment (rcaragon) [reply package.json]: Updated: https://github.com/aragon/ods/pull/215/commits/3e0fcecae25f6a86a9d53ec7585b348a36a239c8

File: .babelrc.json [hunk 2]
```diff
@@ -10,7 +10,12 @@
       }
     ],
     "@babel/preset-typescript",
-    "@babel/preset-react"
+    [
+      "@babel/preset-react",
+      {
+        "runtime": "automatic"
+      }
+    ]
```
Comment (thekidnamedkd) [inline .babelrc.json:18]: What is the case for setting runtime explicitly here? I believe automatic is the default

File: .storybook/main.ts [hunk 3]
```diff
@@ -3,16 +3,22 @@ import type { RuleSetRule } from 'webpack';
...
     stories: ['../docs/**/*.@(md|mdx)', '../src/**/*.stories.@(js|jsx|ts|tsx)', '../src/**/*.@(md|mdx)'],
+
     addons: [
         '@storybook/addon-links',
         '@storybook/addon-essentials',
         { name: '@storybook/addon-styling', options: { postCss: true } },
         '@storybook/addon-designs',
+        '@storybook/addon-mdx-gfm',
+        '@storybook/addon-webpack5-compiler-babel',
+        '@chromatic-com/storybook',
```
Comment (thekidnamedkd) [inline .storybook/main.ts]: As Ruggero mentioned, let's part this out for now. Would love to have a spike on its benefits as I see that Storybook has visual testing built in now with the 8 upgrade. That's exciting!

File: .storybook/main.ts [hunk 4]
```diff
@@ -32,6 +38,8 @@ const config: StorybookConfig = {
 
         return webpackConfig;
     },
+
+    docs: {},
```
Comment (thekidnamedkd) [inline .storybook/main.ts:41]: I believe this comes as default even if not included essentially generating:

 ```ts
 docs: {
        autodocs: 'tag'
 },
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -14,6 +14,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 -   Update minor and patch NPM dependencies
 -   Update `yarn` version to `4.3.0`
 -   Update unit tests to use userEvent instead of fireEvent
+-   Upgrade storybook to v8
```
Comment (thekidnamedkd) [inline CHANGELOG.md:17]: 🙏 nice :)

File: .storybook/preview.tsx [hunk 6]
```diff
@@ -54,6 +54,8 @@ const preview: Preview = {
             </OdsModulesProvider>
         ),
     ],
+
+    tags: ['autodocs'],
```
Comment (thekidnamedkd) [inline .storybook/preview.tsx:58]: By setting this here, can we remove it from each individual story setup?

**Review by thekidnamedkd — COMMENTED** [review]
@rcaragon thanks for the implementation. Excited for this. Left a few questions and comments. ✨

File: .babelrc.json (hunk 2 above)
Comment (rcaragon) [reply .babelrc.json:18]: That's a weird one, without explicitly setting runtime to `automatic` it results in all the stories crashing
image

File: .storybook/preview.tsx (hunk 6 above)
Comment (rcaragon) [reply .storybook/preview.tsx:58]: Removed here https://github.com/aragon/ods/pull/215/commits/2d54d377255723f4e35e6055c46210cc7261969f

**Review by thekidnamedkd — APPROVED** [review]

**Comment by thekidnamedkd** [comment]
image

Let's keep an eye on these addons dependencies. It seems non-blocking for now.

**Comment by cgero-eth** [comment]
> image
> Let's keep an eye on these addons dependencies. It seems non-blocking for now.

Good point @thekidnamedkd, @rcaragon could you have a look at this before merging the PR?

File: package.json [hunk 7]
```diff
@@ -54,6 +54,7 @@
     "@radix-ui/react-tabs": "^1.0.0",
     "@radix-ui/react-toggle-group": "^1.0.0",
     "@radix-ui/react-tooltip": "^1.0.0",
+    "@storybook/addon-styling-webpack": "^1.0.0",
```
[the remark this answers is not in the document]
Comment (rcaragon) [reply package.json]: fixed https://github.com/aragon/ods/pull/215/commits/b775b03565a6641aaa4b37dec31c23de2a1d405b

**Comment by rcaragon** [comment]
> Awesome work. I'm still seeing the warning about github flavored markdown package as a migration tool. Again, non-blocking but there's a migration guide for this as well?

It appears that we don't need that addon, removed here: https://github.com/aragon/ods/pull/215/commits/adf7621933e16604faa0e428a56d668b9c6619c4

**Review by cgero-eth — APPROVED** [review]
Awesome work, thank you both @rcaragon & @thekidnamedkd!

**Pull request merged**